flocker/acceptance/testtools.py

```diff
@@ -936,9 +936,21 @@ def nodes_available():
         ).write()
 
         def failed_query(failure):
-            Message.new(message_type="acceptance:is_available_error",
-                        reason=unicode(failure),
-                        exception=unicode(failure.__class__)).write()
+            try:
+                # Try to detect _WrapperException.
+                reasons = failure.value.reasons
+            except AttributeError:
+                # Guess it was something else.  Do some simpler logging.
+                system = "acceptance:is_available_error"
+                Message.new(
+                    message_type=system,
+                    reason=unicode(failure),
+                    exception=unicode(failure.__class__)
+                ).write()
+            else:
+                # It is one of those.  Log all of the stuff from inside it.
+                for reason in reasons:
+                    write_failure(reason, None, system=system)
```

Comment:
Latest version of Eliot doesn't require `system` argument anyway for `write_failure`, so you can just omit it.